test/protocol/test/textDocument/semanticTokens/full.test.ts

```diff
@@ -0,0 +1,40 @@
+import { expect } from 'chai'
+import { test } from 'mocha'
+import { TestLanguageClient } from '../../../src/TestLanguageClient'
+import { getInitializedClient } from '../../client'
+import { getProjectPath } from '../../helpers'
+import { toUri } from '../../../src/helpers'
+
+let client!: TestLanguageClient
+
+describe('[hardhat] semanticTokens/full', () => {
+  let testPath: string
+
+  before(async () => {
+    client = await getInitializedClient()
+
+    testPath = getProjectPath('hardhat/contracts/semanticTokens/full/SemanticTokens.sol')
+
+    await client.openDocument(testPath)
+  })
+
+  after(async () => {
+    client.closeAllDocuments()
+  })
+
+  test('provides highlighting for types, keywords, functions, numbers and strings', async function () {
+    const semanticTokens = await client.getSemanticTokensFull(toUri(testPath))
+
+    expect(semanticTokens).to.deep.equal({
+      data: [
```

Comment:
Ah. I take it the client-side doesn't obviously expose any decoding functionality?